backend/webdav/webdav.go

```diff
@@ -1184,24 +1320,29 @@ func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, op
 			}
 		}
 	}
+
+
 	err = o.fs.pacer.CallNoRetry(func() (bool, error) {
 		resp, err = o.fs.srv.Call(ctx, &opts)
 		return o.fs.shouldRetry(resp, err)
 	})
+
 	if err != nil {
-		// Give the WebDAV server a chance to get its internal state in order after the
-		// error.  The error may have been local in which case we closed the connection.
-		// The server may still be dealing with it for a moment. A sleep isn't ideal but I
-		// haven't been able to think of a better method to find out if the server has
-		// finished - ncw
-		time.Sleep(1 * time.Second)
-		// Remove failed upload
-		_ = o.Remove(ctx)
-		return err
+		o.fs.root = ""
+		rmdirerr := o.fs.Rmdir(ctx, uploadDir)
```

Comment:
This deleting the upload directory could be called via defer and the OnExit framework - see here for an example

https://github.com/rclone/rclone/blob/9ea990d5a29fb74fbb28855577aa77e0e3c1ceb9/backend/s3/s3.go#L2262-L2274

When rclone supports resuming uploads this will have to change...

Reply:
✅